daemon/SecretsImpl/dataprotector_p.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2018 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#ifndef SAILFISHSECRETS_APIIMPL_DATAPROTECTOR_P_H
+#define SAILFISHSECRETS_APIIMPL_DATAPROTECTOR_P_H
+
+#include <QtCore/QObject>
+#include <QtCore/QString>
+#include <QtCore/QByteArray>
+
+namespace Sailfish {
+
+namespace Secrets {
+
+namespace Daemon {
+
+namespace ApiImpl {
+
+class DataProtector final : public QObject
+{
+
+    Q_OBJECT
+
+public:
+    enum Status
+    {
+        Success = 0,
+        Irretrievable,
+        ErrorGotNullptr,
+        ErrorCannotCreateDirectory,
+        ErrorCannotCreateFile,
+        ErrorCannotOpenFile,
+        ErrorCannotWriteFile,
+        ErrorInvalidPath,
+        ErrorCannotRenameFile,
+        ErrorCannotDeleteFile,
+
+    };
+    Q_ENUM(Status)
+
+    explicit DataProtector(const QString &path, QObject *parent = Q_NULLPTR);
```

Comment:
Just a question, or suggestion, would it be clearer that path should be a directory if the constructor argument would be a QDir?